server/datastore/mysql/microsoft_mdm_test.go

```diff
@@ -67,3 +69,109 @@ func testMDMWindowsEnrolledDevice(t *testing.T, ds *Datastore) {
 	err = ds.MDMWindowsDeleteEnrolledDevice(ctx, enrolledDevice.MDMHardwareID)
 	require.ErrorAs(t, err, &nfe)
 }
+
+func testMDMWindowsPendingCommand(t *testing.T, ds *Datastore) {
+	ctx := context.Background()
+
+	deviceID := uuid.New().String()
+	commandUUID := uuid.New().String()
+	pendingCmd := &fleet.MDMWindowsPendingCommand{
+		CommandUUID:  commandUUID,
+		DeviceID:     deviceID,
+		CmdVerb:      fleet.CmdGet,
+		SettingURI:   "./test/uri",
+		SettingValue: "testdata",
+		DataType:     2,
+		SystemOrigin: false,
+	}
+
+	err := ds.MDMWindowsInsertPendingCommand(ctx, pendingCmd)
+	require.NoError(t, err)
+
+	var ae fleet.AlreadyExistsError
+	err = ds.MDMWindowsInsertPendingCommand(ctx, pendingCmd)
+	require.ErrorAs(t, err, &ae)
+
+	// Getting pending commands first
+	gotPendingCmds, err := ds.MDMWindowsGetPendingCommands(ctx, deviceID)
+	require.NoError(t, err)
+	require.NotZero(t, gotPendingCmds)
+	require.NotZero(t, gotPendingCmds[0].CreatedAt)
+	require.Equal(t, gotPendingCmds[0].DeviceID, deviceID)
+
+	// Now inserting commands in the tracking table
+	newCmd1 := &fleet.MDMWindowsCommand{
+		CommandUUID:  commandUUID,
+		DeviceID:     deviceID,
+		SessionID:    "2",
+		MessageID:    "3",
+		CommandID:    "4",
+		CmdVerb:      fleet.CmdGet,
+		SettingURI:   "./test/uri",
+		SettingValue: "testdata",
+		SystemOrigin: false,
+	}
+
+	newCmd2 := &fleet.MDMWindowsCommand{
+		CommandUUID:  uuid.New().String(),
+		DeviceID:     deviceID,
+		SessionID:    "6",
+		MessageID:    "7",
+		CommandID:    "8",
+		CmdVerb:      fleet.CmdGet,
+		SettingURI:   "./test/uri",
+		SettingValue: "testdata",
+		SystemOrigin: false,
+	}
+
+	err = ds.MDMWindowsInsertCommand(ctx, newCmd1)
+	require.NoError(t, err)
+
+	err = ds.MDMWindowsInsertCommand(ctx, newCmd2)
+	require.NoError(t, err)
+
+	gotCmds, err := ds.MDMWindowsListCommands(ctx, deviceID)
+	require.NoError(t, err)
+	require.NotZero(t, gotCmds)
+
+	// Now checking if pendings table returns empty as commands are inserted in tracking table
+	gotPendingCmds, err = ds.MDMWindowsGetPendingCommands(ctx, deviceID)
```

Comment:
You should probably add 2 commands in the pending table and add non-pending for just one of those, then check that getting the pending returns the only one still pending.

Reply:
good point, just improved the test to cover that